src/esw/brushed_motors/Controller.cpp

```diff
@@ -196,12 +198,42 @@ void Controller::turnOn() const {
 }
 
 // REQUIRES: nothing
-// MODIFIES: isLive
+// MODIFIES: nothing
+// EFFECTS: Returns a combined ID for both the deviceAddress and motorID
+// MotorID can only be max 3 bits (0-5), and device address is max 2 bits (1 or 2)
+uint8_t Controller::combineDeviceMotorID() const {
+    return (deviceAddress << 3) | motorID;
+}
+
+// REQUIRES: nothing
+// MODIFIES: liveMap
 // EFFECTS: I2C bus, if not already live,
 // configures the physical controller.
 // Then makes live.
+
+std::unordered_map<uint8_t, LiveState> Controller::liveMap;
+
 void Controller::makeLive() {
-    if (isLive) {
+    // if (isLive) {
+    //     return;
+    // }
+
+    uint8_t key = combineDeviceMotorID();
+
+    // if key is absent, no motor with that key has been made live
+    auto it = liveMap.find(key);
+    if (it == liveMap.end()) {
+        LiveState state;
+        state.jointName = name;
+        liveMap.emplace(key, state);
```

Comment:
you should use a lock guard whenever emplacing (basically just need to put a lock and unlock around the emplace, but a lock_guard can do that.(something like std::lock_guard<std::mutex> lock(state.liveMutex);  )